core/src/main/java/io/questdb/cairo/wal/WalPurgeJob.java

```diff
@@ -362,10 +404,18 @@ private void discoverWalSegments() {
                                     if (type == Files.DT_DIR && matchesSegmentName(walName.of(pUtf8NameZ))) {
                                         try {
                                             int segmentId = Numbers.parseInt(walName);
+                                            if ((segmentId < WalUtils.SEG_MIN_ID) || (segmentId > WalUtils.SEG_MAX_ID)) {
+                                                throw NumericException.INSTANCE;
+                                            }
                                             Path segmentPath = path.trimTo(walPathLen).slash().put(segmentId);
                                             TableUtils.lockName(segmentPath);
-                                            boolean unlocked = unlocked(segmentPath.$());
-                                            onDiskWalIDSegmentIDPairs.add(Numbers.encodeLowHighInts((segmentId << 1) + (unlocked ? 0 : 1), walId));
+                                            boolean locked = !unlocked(segmentPath.$());
+                                            boolean pendingTasks = segmentHasPendingTasks(walId, segmentId);
+                                            if (pendingTasks) {
+                                                pendingWalIDSet.add(walId);
```

Comment:
Why do we need segments with pending tasks in the result set?
Can we do just 

```suggestion
                                                pendingWalIDSet.add(walId);
                                                continue;
```

 here?

Reply:
Absolutely. It will simplify the bit packing logic too, and revert it to what it was before. Thanks for the idea. I'll change it.

Reply:
I had started making the change, but on a second thought I think the code should stay as it is:

The reason is this ease of debugging. See:

```java
boolean tableDropped = false;
discoverWalSegments();  // <--- this would skip onDiskWalIDSegmentIDPairs
if (onDiskWalIDSegmentIDPairs.size() != 0) {
    tableDropped = fetchSequencerPairs();
    accumDebugState(); // <--- thus this would not log.
```

If I never add it to the set, it will never shop up in the "accumDebugState".
I think this would hinder debuggability, so the current version of the code (whilst slightly more complicated) is the one that I prefer.

@ideoma, what do you think?

Reply:
Based on conversation on Slack, will keep the existing code as is for better debuggability.